src/commands/bootstrap/index.ts

```diff
@@ -0,0 +1,685 @@
+    onState: onPromptState,
+    type: 'select',
+    name: 'framework',
+    message: `What framework would you like to use?`,
+    choices: frameworkOptions,
+    initial: 0,
+  });
+  finalOptions.framework = frameworkOptions[framework]
+    .title as typeof finalOptions.framework;
+  trackEvent('create-app', {
+    phase: 'framework',
+    meta: { framework: finalOptions.framework },
+  });
+
+  const { orm } = await prompts({
+    onState: onPromptState,
+    type: 'select',
+    name: 'orm',
+    message: `What ORM would you like to use?`,
+    choices: [
+      { title: 'Drizzle', value: 'drizzle' },
+      { title: 'Prisma', value: 'prisma', disabled: true },
+      { title: 'No ORM', value: -1, disabled: true },
+    ],
+    initial: 0,
+  });
+  finalOptions.orm = orm;
+  trackEvent('create-app', { phase: 'orm', meta: { orm: finalOptions.orm } });
+
+  const { auth } = await prompts({
+    onState: onPromptState,
+    type: 'select',
+    name: 'auth',
+    message: `What authentication framework do you want to use?`,
+    choices: [
+      { title: 'Auth.js', value: 'auth.js' },
+      { title: 'No Authentication', value: -1 },
+    ],
+    initial: 0,
+  });
+  finalOptions.auth = auth;
+  trackEvent('create-app', {
+    phase: 'auth',
+    meta: { auth: finalOptions.auth },
+  });
+
+  const PROJECTS_LIST_LIMIT = 100;
+  const getList = async (
+    fn:
+      | typeof props.apiClient.listProjects
+      | typeof props.apiClient.listSharedProjects,
+  ) => {
+    const result: ProjectListItem[] = [];
+    let cursor: string | undefined;
+    let end = false;
+    while (!end) {
+      const { data } = await fn({
+        limit: PROJECTS_LIST_LIMIT,
+        cursor,
+      });
+      result.push(...data.projects);
+      cursor = data.pagination?.cursor;
+      log.debug(
+        'Got %d projects, with cursor: %s',
+        data.projects.length,
+        cursor,
+      );
+      if (data.projects.length < PROJECTS_LIST_LIMIT) {
+        end = true;
+      }
+    }
+
+    return result;
+  };
+
+  const [ownedProjects, sharedProjects] = await Promise.all([
+    getList(props.apiClient.listProjects),
+    getList(props.apiClient.listSharedProjects),
+  ]);
+  const allProjects = [...ownedProjects, ...sharedProjects];
+
+  const projectChoices = [
+    { title: 'Create a new Neon project', value: -1 },
+    ...allProjects.map((project) => {
+      return {
+        title: project.name,
+        value: project.id,
+      };
+    }),
+  ];
+
+  // `neonProject` will either be -1 or a string (project ID)
+  const { neonProject } = await prompts({
+    onState: onPromptState,
+    type: 'select',
+    name: 'neonProject',
+    message: `What Neon project would you like to use?`,
+    choices: projectChoices,
+    initial: 0,
+  });
+  trackEvent('create-app', { phase: 'neon-project' });
+
+  let project;
+  let connectionString: string;
+  if (neonProject === -1) {
+    project = await create({ ...props, psql: false, setContext: false });
```

Comment:
I'm not sure what the benefit is of not calling the create API method directly. Also, this method will throw if the project limit is reached (IIRC)

Reply:
I like that this method prints nice useful information to the console. Yes, it will throw if `props.apiClient.createProject` throws, which is also a good thing. I've wrapped it in a try/catch now!